DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Layers/VectorLayerToHostConverter.cs

```diff
@@ -0,0 +1,110 @@
+
+  public VectorLayerToHostConverter(
+    IConversionContextStack<Map, Unit> contextStack,
+    IRawConversion<Base, ArcGIS.Core.Geometry.Geometry> gisGeometryConverter
+  )
+  {
+    _contextStack = contextStack;
+    _gisGeometryConverter = gisGeometryConverter;
+  }
+
+  public object Convert(Base target) => RawConvert((VectorLayer)target);
+
+  public Task<string> RawConvert(VectorLayer target)
+  {
+    try
+    {
+      return QueuedTask.Run(() =>
+      {
+        var projectUtils = new ArcGISProjectUtils();
+        var utils = new FeatureClassUtils();
+
+        string databasePath = projectUtils.GetDatabasePath();
+        FileGeodatabaseConnectionPath fileGeodatabaseConnectionPath = new(new Uri(databasePath));
+        Geodatabase geodatabase = new(fileGeodatabaseConnectionPath);
+        SchemaBuilder schemaBuilder = new(geodatabase);
+
+        // https://pro.arcgis.com/en/pro-app/3.1/tool-reference/tool-errors-and-warnings/001001-010000/tool-errors-and-warnings-00001-00025-000020.htm
+        string featureClassName = $"{target.id}___{target.name.Replace(" ", "_").Replace("%", "_").Replace("*", "_")}";
+
+        string wktString = string.Empty;
+        if (target.crs is not null && target.crs.wkt is not null)
+        {
+          wktString = target.crs.wkt.ToString();
+        }
+        SpatialReference spatialRef = SpatialReferenceBuilder.CreateSpatialReference(wktString);
+
+        GeometryType geomType = utils.GetLayerGeometryType(target);
+
+        // Create FeatureClass
+        List<FieldDescription> fields = new();
+        List<string> fieldAdded = new();
+        foreach (var field in target.attributes.GetMembers(DynamicBaseMemberType.Dynamic))
+        {
+          if (!fieldAdded.Contains(field.Key) && field.Key != "OBJECTID")
+          {
+            // TODO: choose the right type for Field
+            // TODO check for the frbidden characters/combinations: https://support.esri.com/en-us/knowledge-base/what-characters-should-not-be-used-in-arcgis-for-field--000005588
+
+            // fields.Add(new FieldDescription(field, FieldType.Integer));
+            fields.Add(FieldDescription.CreateStringField(field.Key, 255)); // (int)(long)target.attributes[field.Value]));
+            fieldAdded.Add(field.Key);
+          }
+        }
+        try
+        {
+          FeatureClassDescription featureClassDescription =
+            new(featureClassName, fields, new ShapeDescription(geomType, spatialRef));
+          FeatureClassToken featureClassToken = schemaBuilder.Create(featureClassDescription);
+        }
+        catch (ArgumentException ex)
+        {
+          // if name has invalid characters/combinations
+          throw new ArgumentException($"{ex.Message}: {featureClassName}");
+        }
+
+        bool buildStatus = schemaBuilder.Build();
+        if (!buildStatus)
+        {
+          IReadOnlyList<string> errors = schemaBuilder.ErrorMessages;
```

Comment:
Not sure what's happening here, given the errors immediately go out of scope, perhaps include a POC comment